extensions/blocks/simple-payments/simple-payments.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Pay with PayPal block (aka Simple Payments).
+ *
+ * @since 9.0.0
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\SimplePayments;
+
+use Jetpack_Simple_Payments;
+
+const FEATURE_NAME = 'simple-payments';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	jetpack_register_block(
+		BLOCK_NAME,
+		array(
+			'render_callback' => __NAMESPACE__ . '\render_block',
+			'plan_check'      => true,
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Pay with PayPal block dynamic rendering.
+ *
+ * @param array  $attr    Array containing the block attributes.
+ * @param string $content String containing the block content.
+ *
+ * @return string
+ */
+function render_block( $attr, $content ) {
+	// Do nothing if block content is a `simple-payment` shortcode.
+	if ( preg_match( '/\[simple-payment(.*)]/', $content ) ) {
+		return $content;
+	}
+
+	// Keep content as-is if rendered in other contexts than frontend (i.e. feed, emails, API, etc.).
+	if ( ! jetpack_is_frontend() ) {
```

Comment:
I wonder if we should add a `Blocks::is_amp_request()` check here as well, since this may be another context where we'd want to display a simplified block.

cc @westonruter What are your thoughts on this?

Reply:
Yes, this makes sense to me. I assume the button just doesn't work in AMP currently, so serving the fallback static content which we know is AMP-compatible seems correct.

Reply:
```suggestion
	if ( ! jetpack_is_frontend() || Blocks::is_amp_request() ) {
```

I'm not set up to test this block, but what does this result in on AMP pages?

Reply:
https://github.com/Automattic/jetpack/pull/17298/